reactive/kubernetes_worker.py

```diff
@@ -900,39 +909,61 @@ def get_kube_api_servers(kube_api):
 
 @when('nrpe-external-master.available')
 @when_not('nrpe-external-master.initial-config')
-def initial_nrpe_config(nagios=None):
+def initial_nrpe_config():
     set_state('nrpe-external-master.initial-config')
-    update_nrpe_config(nagios)
+    update_nrpe_config()
 
 
 @when('kubernetes-worker.config.created')
 @when('nrpe-external-master.available')
+@when('kube-api-endpoint.available')
+@when('kube-control.auth.available')
 @when_any('config.changed.nagios_context',
-          'config.changed.nagios_servicegroups')
-def update_nrpe_config(unused=None):
-    services = ('snap.kubelet.daemon', 'snap.kube-proxy.daemon')
+          'config.changed.nagios_servicegroups',
+          'nrpe-external-master.reconfigure')
+def update_nrpe_config():
+    services = ['snap.{}.daemon'.format(s) for s in worker_services]
+    data = render('nagios_plugin.py', context={'node_name': get_node_name()})
+    plugin_path = install_nagios_plugin_from_text(data,
+                                                  'check_k8s_worker.py')
     hostname = nrpe.get_nagios_hostname()
     current_unit = nrpe.get_nagios_unit_name()
     nrpe_setup = nrpe.NRPE(hostname=hostname)
+    nrpe_setup.add_check("node",
+                         "Node registered with API Server",
+                         str(plugin_path))
     nrpe.add_init_service_checks(nrpe_setup, services, current_unit)
     nrpe_setup.write()
 
+    creds = db.get('credentials')
+    if creds:
+        kube_api = endpoint_from_flag('kube-api-endpoint.available')
+        servers = get_kube_api_servers(kube_api)
+        server = servers[get_unit_number() % len(servers)]
+        create_kubeconfig(nrpe_kubeconfig_path, server, ca_crt_path,
+                          token=creds['client_token'], user='nagios')
+        # Make the config dir owned by the nagios user.
+        cmd = ['chown', '-R', 'nagios:nagios',
+               os.path.dirname(nrpe_kubeconfig_path)]
+        check_call(cmd)
+
+    remove_state('nrpe-external-master.reconfigure')
```

Comment:
you sure we always want to remove the `reconfigure` flag here?  seems like if we don't have creds, we'd want to retry `update_nrpe_config` until we do.

Reply:
I get where you are coming from, but if we don't have credentials when we get them we restart the worker, which causes us to set nrpe-external-master.reconfigure again.

Reply:
that feels like happenstance.  if we don't have creds, we remove the reconfig state and hope that the next reactive iteration sets enough bits to enter this handler.  fragile++

Reply:
By the same token, if we don't remove it we spam call update_nrpe_config until creds become available, which is going to call us again anyway. I'm not sure which way to fall on this one, so I'll let it be yours with sufficient comments in the git history so we know who to blame...

Reply:
If we remove `.reconfigure` here when we don't have creds, we may not re-enter this handler because `.reconfigure` (one of the of `when_any's`) won't be set. I'd rather spam this handler to ensure our nagios kubeconfig is legit rather than missing the essential `create_kubeconfig(...nagios...)`.

Let's pick this up in #42.